sql/analyzer/replace_pk_sort.go

```diff
@@ -123,3 +142,143 @@ func replacePkSortHelper(ctx *sql.Context, scope *plan.Scope, node sql.Node, sor
 	}
 	return newNode, transform.NewTree, nil
 }
+
+// replaceAgg converts aggregate functions to order by + limit 1 when possible
+func replaceAgg(ctx *sql.Context, a *Analyzer, node sql.Node, scope *plan.Scope, sel RuleSelector) (sql.Node, transform.TreeIdentity, error) {
+	return transform.Node(node, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
+		// project with groupby child
+		proj, ok := n.(*plan.Project)
+		if !ok {
+			return n, transform.SameTree, nil
+		}
+		gb, ok := proj.Child.(*plan.GroupBy)
+		if !ok {
+			return n, transform.SameTree, nil
+		}
+		// TODO: optimize when there are multiple aggregations; use LATERAL JOINS
+		if len(gb.SelectedExprs) != 1 || len(gb.GroupByExprs) != 0 {
+			return n, transform.SameTree, nil
+		}
+
+		var idxTbl sql.IndexAddressableTable
+		transform.Inspect(gb.Child, func(n sql.Node) bool {
+			switch t := gb.Child.(type) {
+			case *plan.IndexedTableAccess:
```

Comment:
a LOOKUP_JOIN might be a false positive here, indexed table but values won't be in order